hydromt/raster.py

```diff
@@ -1651,6 +1655,110 @@ def interpolate_na(self, method: str = "nearest", **kwargs):
         )
         return interp_array
 
+    def to_xyz(
+        self,
+        root: str,
+        px: int,
+        zoomlevels: list = [],
+    ):
+        """Export rasterdataset to tiles in a xyz structure
+
+        Parameters
+        ----------
+        root : str
+            Path where the database will be saved
+            Database yml will be put one directory above
+        px : int
+            Amount of pixels per tile
+        zoomlevels : list, optional
+            Zoom levels to be put in the database
+
+        Raises
+        ------
+        ValueError
+            Amount of pixels can not be larger than the original dataset
+        """
+
+        mName = os.path.normpath(os.path.basename(root))
+
+        def folder(path):
+            if not os.path.exists(path):
+                os.makedirs(path)
+
+        folder(root)
+
+        mdim = min(self._obj.shape)
+        if px > mdim:
+            raise ValueError("")
+        nzl = int(np.ceil((np.log10(mdim / px) / np.log10(2))))
+        pxs = [px * 2**num for num in range(nzl)]
+
+        # Clean up zoomlevels list
+        zoomlevels = [x for x in zoomlevels if x <= len(pxs) - 1]
+
+        def tile_window(px):
+            # Basic stuff
+            nr, nc = self._obj.shape
+            lu = product(range(0, nc, px), range(0, nr, px))
+            ## create the window
+            for l, u in lu:
+                h = min(px, nr - u)
+                w = min(px, nc - l)
+                yield (l, u, w, h)
+
+        for zl in zoomlevels:
+            sd = f"{root}\\{zl}"
+            folder(sd)
+            # Write the raster paths to a text file
+            file = open(f"{sd}\\tilelist.txt", "w")
+
+            for l, u, w, h in tile_window(pxs[zl]):
+                col = int(np.ceil(l / pxs[zl]))
+                row = int(np.ceil(u / pxs[zl]))
+                ssd = "{}\\{}".format(sd, col)
+
+                folder(ssd)
+
+                # create temp tile
+                temp = self._obj[u : u + h, l : l + w].load()
+                # temp = self._obj.isel({self.y_dim:slice(u,u+h), self.x_dim: slice(l,l+w)})
+                if zl != 0:
+                    dst_transform = temp.raster.transform * temp.raster.transform.scale(
+                        2**zl
+                    )
+                    temp = temp.raster.reproject(
+                        dst_transform=dst_transform,
+                        dst_width=int(w / (2**zl)),
+                        dst_height=int(h / (2**zl)),
+                        method="average",
+                        # dst_crs=self.crs,
+                    )
+
+                temp.raster.to_raster(f"{ssd}\\{mName}_{col}_{row}.tif", driver="GTiff")
+
+                file.write(f"{ssd}\\{mName}_{col}_{row}.tif\n")
+
+                del temp
+
+            file.close()
+            # Create a vrt using GDAL
```

Comment:
I think it be useful to make this into a separate `buildvrt` function in gis_utils.py

Reply:
Done